### PR TITLE
Update sample for Microsoft Graph

### PR DESCRIPTION
**Pull request by rgregg**

### CODE REVIEW

File: debug.log [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[0205/143655:ERROR:tcp_listen_socket.cc(76)] Could not bind socket to 127.0.0.1:6004
+[0205/143655:ERROR:node_debugger.cc(86)] Cannot start debugger server
```
Comment (daspek) [inline debug.log]: what's this file?

File: index.html [hunk 2]
```diff
@@ -1,21 +1,144 @@
...
+  <link rel="icon" href="https://d.sfx-df.ms/images/favicon.ico" type="image/x-icon">
+  <link rel="mask-icon" href="https://d.sfx-df.ms/images/mask_icon.svg" color="#094AB2">
+  <script>
+    function signInToOneDrive()
+    {
+      // Register your own application at https://apps.dev.microsoft.com
+      // and set the "clientId" and "redirectUri" variables accordingly.
+      var appInfo = {
+        "clientId": "e9a57c07-69e4-41f5-b868-0f123d2fda17",
+        "redirectUri": "http://localhost:9999/callback.html",
```
Comment (daspek) [inline index.html:17]: this needs to be the real ODX one, right?

File: index.html [hunk 3]
```diff
@@ -191,16 +318,16 @@
...
-                else {
+                else if (data.file) {
                   // 1-up view
                   var tile = $("<div>").
                     addClass("item").
                     addClass("oneup").
                     appendTo("#od-items");
 
-                  var downloadUrl = data['@content.downloadUrl'];
+                  var downloadUrl = data['webUrl'];
```
Comment (daspek) [inline index.html]: that's not the download url though

File: odauth.js [hunk 4]
```diff
@@ -55,15 +55,18 @@ function onAuthCallback() {
...
+    window.opener.onAuthenticated(token, window);
+  }
 }
 
 function getAuthInfoFromUrl() {
   if (window.location.hash) {
     var authResponse = window.location.hash.substring(1);
     var authInfo = JSON.parse(
-      '{"' + authResponse.replace(/&/g, '","').replace(/=/g, '":"') + '"}',
+      '{' + authResponse.replace(/([^=]+)=([^&]+)&?/g, '"$1":"$2",').slice(0,-1) + '}',
```
Comment (daspek) [inline odauth.js:69]: oh yeah....

File: odauth.js [hunk 5]
```diff
@@ -117,26 +138,37 @@ function getAppInfo() {
   }
 
   var scopes = scriptTag.getAttribute("scopes");
-  if (!scopes) {
-    alert("the odauth script tag needs a scopes attribute set to the scopes your app needs");
-  }
+  // if (!scopes) {
```
Comment (daspek) [inline odauth.js]: gonna remove?

**Review by daspek — APPROVED** [review]

File: odauth.js (hunk 5 above)
Comment (rgregg) [reply odauth.js]: You left it there :).

File: index.html (hunk 3 above)
Comment (rgregg) [reply index.html]: Oops, good catch.

File: index.html (hunk 2 above)
Comment (rgregg) [reply index.html:17]: The live version uses a copy of this code, and does't pull from the repo. It has the right values for the live version.

**Pull request merged**